docs/narr/advfeatures.rst

```diff
@@ -0,0 +1,399 @@
+Advanced :app:`Pyramid` Design Features
+=======================================
+
+Pyramid has been built from the ground up to avoid the problems that other
+frameworks can suffer.
+
+
+You Don't Need Singletons
+-------------------------
+
+Have you ever struggled with parametrizing Django's ``settings.py`` file for
+multiple installations of the same Django application? Have you ever needed to
+monkey-patch a framework fixture to get it to behave properly for your
+use-case? Have you ever tried to deploy your application using an asynchronous
+server and failed?
+
+All these problems are symptoms of "mutable global state", also known as
+"import-time side effects" and arise from the use of "singleton" data structures.
```

Comment:
I've added some of these to the glossary and converted them to `:term:` references